fuel/transformers/__init__.py

```diff
@@ -60,6 +61,24 @@ def get_epoch_iterator(self, **kwargs):
         self.child_epoch_iterator = self.data_stream.get_epoch_iterator()
         return super(Transformer, self).get_epoch_iterator(**kwargs)
 
+    def get_data(self, request=None):
+        if self.batch_input:
+            return self.get_data_from_batch(request)
+        else:
+            return self.get_data_from_example(request)
+
+    def get_data_from_example(self, request=None):
```

Comment:
Ah, this is something I discussed with @mducoffe by e-mail. Let me just copy-paste the rationale. They're really just renames of the methods we had for these reasons:

> I've made some comments, but I realised that the naming is awkward though... Right now get_example and get_batch refer to what the *output* is (as their name suggests) but the difference between get_example and get_batch should really depend on what the *input* is. Technically speaking you could have a transformer that can take either an example or a batch as an input, but will output a batch in both cases. The methods will still need to be different though.

> A very artificial example, but consider a transformer which outputs batches of size N. Its behaviour changes depending on the input, but the output is always the same:
>
```python
class BatchOfNExamples(Transformer):
    def get_data_from_batch(self):
        batch = []
        while len(batch) < N:
            batch += next(self.child_epoch_iterator)[:N - len(batch)]
        return batch
>
    def get_data_from_example(self):
        return [next(self.child_epoch_iterator) for _ in range(N)]
```

> Maybe a good idea would then be to rename get_batch to get_data_from_batch and get_example to get_data_from_example (means you'll have to swap them around for Unpack and Batch), and change the keyword argument from `batch` to `batch_input` to make clear that it refers to what the input is, not to what the output is. (I hope you're familiar with your text editor's renaming function...)